### DISCUSSION TITLE
Adding multiple PV arrays

### DISCUSSION THREAD

**Discussion by hamishbeath** [CLOVER-energy/CLOVER#198] [IAA]
### Issue
At present, the model takes input for the PV to be fed to renewable.ninja. The inputs relate to orientation, tilt etc. Renewables.ninja returns expected solar output for that PV type. The model can only accept one PV array with a given configuration. However, if simulating an existing system it would be good to be able to include multiple arrays with different orientations.

The issue that you are facing and would like fixed or implemented.
It would be good to have a change implemented for simulation functioning that permits the addition of more than one PV array. with different orientation/tilt. Longer term, outlining orientation, angle (and possibly space available) for PV arrays for the optimisation function may be a desirable feature.

### Proposal
When running a simulation with multiple PV arrays that have a different orientation or tilt, additional necessary requests should be made to renewables.ninja. These outputs are stored and used for each array with corresponding size.

**Describe alternatives you've considered**
Leaving as is.

**Comment by BenWinchester** [comment]
@hamishbeath , I like this idea! :tada:

On this, I wonder whether it makes sense to include some functionality re: mesh grids along with this? This should likely be a different issue but I think there are potentially links in how we implement this functionality.

**Comment by BenWinchester** [comment]
## General thoughts
In order to accurately model agri-PV arrays, solar panels at different orientations and made of different materials will need to be modellablae. These range from standard panels (i.e., mono-crystaline Silicon (m-Si) PV panels) already considered within CLOVER, which may need to be at different orientations within the same system to accurately reflect an agri-PV array, to semi-transparrent panels which may need to be in a curved topology as well as multiple different technologies.

## ⬆️ Promotion to an issue ⬆️ 
As this is something that needs to be done to accurately reflect some of the essential features of an agri-PV setup, it will be referenced in a new issue 😄
